### ISSUE TITLE
Scrubber connects to the wrong proxy

### ISSUE THREAD

**Issue by jeroenmaelbrancke** [openvstorage/framework#1999]
# Problem
On SV the scrubber from vpool data-sv-4-02 connects to a proxy of ss-sv4-01.
In the celery log the config that has been used for the proxy is
```
Config: {u'block_cache': [u'none'], u'log_level': u'info', u'ips': [u'127.0.0.1'], u'manifest_cache_size': 524288000, u'fragment_cache': [u'alba', {u'albamgr_cfg_url': u'arakoon://config/ovs/vpools/08f94559-bc58-454b-9cd7-9ec1ff712e16/proxies/d37f9d7f-2ee9-4d7a-b24d-bfcab2f78d39/config/abm_aa?ini=%2Fopt%2FOpenvStorage%2Fconfig%2Farakoon_cacc.ini', u'cache_on_write': True, u'manifest_cache_size': 524288000, u'cache_on_read': False, u'bucket_strategy': [u'1-to-1', {u'prefix': u'08f94559-bc58-454b-9cd7-9ec1ff712e16', u'preset': u'cache_no_encrypt'}]}], u'read_preference': [], u'albamgr_cfg_url': u'arakoon://config/ovs/vpools/08f94559-bc58-454b-9cd7-9ec1ff712e16/proxies/d37f9d7f-2ee9-4d7a-b24d-bfcab2f78d39/config/abm?ini=%2Fopt%2FOpenvStorage%2Fconfig%2Farakoon_cacc.ini', u'port': 26202, u'transport': 'tcp'})
```

I cannot find any  error logs in the celery log that the proxy creation failed.
On The scrub logging the first x volumes failed with Connection refused:

```
2018-03-13 10:29:38 208377 -0700 - SV1SRV0001 - 7955/0x00007fea7de73900 - scrubber/AlbaConnection - 0000000000000017 - error - convert_exceptions_: make proxy client: caught boost system_error: Connection refused
2018-03-13 10:29:38 208425 -0700 - SV1SRV0001 - 7955/0x00007fea7de73900 - scrubber/AlbaConnection - 0000000000000018 - error - convert_exceptions_: assuming proxy connection failure
2018-03-13 10:29:38 208480 -0700 - SV1SRV0001 - 7955/0x00007fea7de73900 - scrubber/BackendConnectionPool - 0000000000000019 - error - make_one_: AlbaConfig{host=127.0.0.1,port=26202,timeout=30,transport=TCP,use_rora=true,rora_manifest_cache_capacity=25000,rora_use_nullio=false,asd_connection_pool_capacity=10,rora_timeout_msecs=50,preset=local_encrypt}: failed to create new connection:BackendException: failed to connect
2018-03-13 10:29:38 208507 -0700 - SV1SRV0001 - 7955/0x00007fea7de73900 - scrubber/BackendConnectionPool - 000000000000001a - warning - error: AlbaConfig{host=127.0.0.1,port=26202,timeout=30,transport=TCP,use_rora=true,rora_manifest_cache_
capacity=25000,rora_use_nullio=false,asd_connection_pool_capacity=10,rora_timeout_msecs=50,preset=local_encrypt}: blacklisted for 15000000000 nanoseconds
2018-03-13 10:29:38 208602 -0700 - SV1SRV0001 - 7955/0x00007fea7de73900 - scrubber/BackendInterface - 000000000000001b - error - do_wrap_: 4cedd7fc-244f-4306-9ebb-ac7b8a9d9f20: connection failure
2018-03-13 10:29:38 208631 -0700 - SV1SRV0001 - 7955/0x00007fea7de73900 - scrubber/BackendInterface - 000000000000001c - warning - do_wrap_: 4cedd7fc-244f-4306-9ebb-ac7b8a9d9f20: retrying with new connection (retry: 1, sleep before retry: 1000 milliseconds)
```
After 5 minutes the other volumes errored with Namespace doesn't exists.
```
2018-03-13 10:32:44 807836 -0700 - SV1SRV0001 - 9907/0x00007f34a9332900 - scrubber/AlbaConnection - 0000000000000069 - error - convert_exceptions_: read object: caught Alba proxy exception: Proxy_protocol.Protocol.Error.NamespaceDoesNotExist
```

So the scrubber was able to connect to A proxy but the proxy was not aware of these volumes...
Checking the alba.logs find out a second proxy was deployed for ss-sv4-01 with the same port!!!!!
How is that possible?
If someone start a scrub process for another vpool this vpool cannot use the port deployed for vpool data-sv4-02.

```
Mar 13 10:32:44 SV1SRV0001 alba[10496]: 2018-03-13 10:32:44 663708 -0700 - SV1SRV0001 - 10496/0000 - alba/proxy - 0 - info - Found the followin
g config: {;    "block_cache": [; "none";    ], ;    "log_level": "info", ;    "fragment_cache": [; "none";    ], ;    "manifest_cache_size": 524288000, ;    "ips": [; "127.0.0.1";   ], ;   "read_preference": [], ;    "albamgr_cfg_url": "arakoon://config/ovs/vpools/a99e685c-46f9-4bd4-9017-5355732c2afb/proxies/cab5e9e6-2938-4ec0-8d59-a4e4b0be0cc7/config/abm?ini=%2Fopt%2FOpenvStorage%2Fconfig2Farakoon_cacc.ini", ;    "port": 26202, ;    "transport": "tcp";}
Mar 13 10:32:44 SV1SRV0001 alba[10496]: 2018-03-13 10:32:44 664801 -0700 - SV1SRV0001 - 10496/0000 - alba/proxy - 1 - info - Interpreted the config as: { Cli_proxy.Config.ips = ["127.0.0.1"]; transport = "tcp"; port = 26202;;  log_level = "info"; albamgr_cfg_file = None;;  albamgr_cfg_url =;  (Some "arakoon://config/ovs/vpools/a99e685c-46f9-4bd4-9017-5355732c2afb/proxies/cab5e9e6-2938-4ec0-8d59-a4e4b0be0cc7/config/abm?ini=%2Fopt%2FOpenvStorage%2Fconfig%2Farakoon_cacc.ini");;  fragment_cache = (Some Blob_cache_config.None');;  block_cache = (Some Blob_cache_config.None'); fragment_cache_dir = None;;  fragment_cache_size = None; manifest_cache_size = 524288000;;  albamgr_connection_pool_size = 10; nsm_host_connection_pool_size = 10;;  osd_connection_pool_size = 10; osd_timeout = 10.;;  lwt_preemptive_thread_pool_min_size = 10;;  lwt_preemptive_thread_pool_max_size = 20; chattiness = None;;  max_client_connections = 128; tls_client = None;;  tcp_keepalive =;  { Tcp_keepalive.enable_tcp_keepalive = true; tcp_keepalive_time = 20;;    tcp_keepalive_intvl = 20; tcp_keepalive_probes = 3 };;  use_fadvise = true; upload_slack = 0.2; read_preference = [];;  propagate_osd_info_delay = 60.; fill_bc_from_fc = false; pracc = None;;  partial_read_retry_count = 2; partial_read_retry_interval_seconds = 0.5;;  partial_read_retry_backoff_multiplier = 2.; partial_reconstruct = true }
```
For data-sv4-02 the proxy was never running at that point.

So how do you guys check if a scrub proxy has been deployed and is running?
Do you only check if the port is online or do you check via the service manager if the process is running?

# Logs

celery.log:
```
Mar 13 10:28:20 SV4SRV0002 celery[32098]: 2018-03-13 10:28:20 68200 -0700 - SV4SRV0002 - 32098/140081803060992 - celery/celery.worker.strategy - 5703 - INFO - Received task: ovs.generic.execute_scrub[c61e49e1-7708-40f7-829d-80150fd9d811]
Mar 13 10:28:20 SV4SRV0002 celery[32098]: 2018-03-13 10:28:20 69000 -0700 - SV4SRV0002 - 34768/140081803060992 - lib/ensure single - 5700 - INFO - Ensure single DEDUPED mode - ID 1520962100_LnUlNX1QAQ - New task ovs.generic.execute_scrub with params {'manual': True, 'vdisk_guids': [u'7eea4242-8876-47f4-b069-a3c20d19a1a7', u'876dd52b-39b0-4a73-8660-dfe94a91c21e', u'fb19b5a1-2054-4bac-b039-028fbd809945',
Mar 13 10:28:24 SV4SRV0002 celery[32098]: 2018-03-13 10:28:24 08700 -0700 - SV4SRV0002 - 34768/140081803060992 - lib/generic tasks scrub - 5702 - INFO - Scrubber c61e49e1-7708-40f7-829d-80150fd9d811 - Storage Router 172.26.23.4 has SCRUB partition at /mnt/hdd1
Mar 13 10:28:31 SV4SRV0002 celery[32098]: 2018-03-13 10:28:31 46100 -0700 - SV4SRV0002 - 34768/140081803060992 - lib/generic tasks scrub - 5703 - INFO - Scrubber c61e49e1-7708-40f7-829d-80150fd9d811 - Executing scrub
Mar 13 10:28:31 SV4SRV0002 celery[32098]: 2018-03-13 10:28:31 46800 -0700 - SV4SRV0002 - 34768/140081803060992 - lib/generic tasks scrub - 5704 - INFO - Scrubber c61e49e1-7708-40f7-829d-80150fd9d811 - vPool data-sv4-02 - Checking scrub work
Mar 13 10:28:31 SV4SRV0002 celery[32098]: 2018-03-13 10:28:31 47200 -0700 - SV4SRV0002 - 34768/140081803060992 - lib/generic tasks scrub - 5705 - INFO - Scrubber c61e49e1-7708-40f7-829d-80150fd9d811 - vPool data-sv4-02 - Will be removing {u'worker_pid': 26225, u'vdisk_guid': u'52457ddd-439e-4254-aa4d-e4b86e16fad1', u'storagerouter_guid': u'26955ac4-d6cb-4abc-9536-bf1b74527b32', u'worker_start': u'Mon Mar  5 13:26:46 2018'} on the next save as it is no longer relevant
...
Mar 13 10:28:31 SV4SRV0002 celery[32098]: 2018-03-13 10:28:31 99900 -0700 - SV4SRV0002 - 34768/140081803060992 - lib/generic tasks scrub - 5852 - INFO - Scrubber c61e49e1-7708-40f7-829d-80150fd9d811 - vPool data-sv4-02 - Following items wi
ll be scrubbed: ['5b2409c2-d2b4-4c83-ba80-2dd923f8106b', '0bf4e158-bbc1-434c-a256-5f8b931fb787', '72bc9ef7-cf8c-440c-bb79-4802000b27ed',
Mar 13 10:28:32 SV4SRV0002 celery[32098]: 2018-03-13 10:28:31 100000 -0700 - SV4SRV0002 - 34768/140081803060992 - lib/generic tasks scrub - 5853 - INFO - Scrubber c61e49e1-7708-40f7-829d-80150fd9d811 - vPool data-sv4-02 - Spawning 1 stack
Mar 13 10:28:32 SV4SRV0002 celery[32098]: 2018-03-13 10:28:32 00600 -0700 - SV4SRV0002 - 34768/140081113306880 - lib/generic tasks scrub - 5854 - INFO - Scrubber c61e49e1-7708-40f7-829d-80150fd9d811 - vPool data-sv4-02 - StorageRouter SV1SRV0001 - Stack 0 - Checking for ALBA proxy ovs-albaproxy_data-sv4-02_SV1SRV0001_568a899a-f674-43ce-9649-6c340321a7e6_scrub deployment
Mar 13 10:28:32 SV4SRV0002 celery[32098]: 2018-03-13 10:28:32 00600 -0700 - SV4SRV0002 - 34768/140081113306880 - lib/generic tasks scrub - 5855 - INFO - Scrubber c61e49e1-7708-40f7-829d-80150fd9d811 - vPool data-sv4-02 - StorageRouter SV1SRV0001 - Stack 0 - Registering usage of ovs-albaproxy_data-sv4-02_SV1SRV0001_568a899a-f674-43ce-9649-6c340321a7e6_scrub
Mar 13 10:28:32 SV4SRV0002 celery[32098]: 2018-03-13 10:28:32 00900 -0700 - SV4SRV0002 - 34768/140081113306880 - lib/generic tasks scrub - 5856 - INFO - Scrubber c61e49e1-7708-40f7-829d-80150fd9d811 - vPool data-sv4-02 - StorageRouter SV1SRV0001 - Stack 0 - Current registered users: [{'stack_id': '514d1ec9-9ff5-447c-991c-9a7a6caa2467', 'storagerouter_guid': '3579dde2-a7c2-4e4f-a24e-295dadd3a0f6', 'job_id': 'c61e49e1-7708-40f7-829d-80150fd9d811', 'worker_pid': 32098, 'worker_start': 'Mon Mar 12 14:19:18 2018'}]
Mar 13 10:28:32 SV4SRV0002 celery[32098]: 2018-03-13 10:28:32 01000 -0700 - SV4SRV0002 - 34768/140081113306880 - lib/generic tasks scrub - 5857 - INFO - Scrubber c61e49e1-7708-40f7-829d-80150fd9d811 - vPool data-sv4-02 - StorageRouter SV1SRV0001 - Stack 0 - Deploying ALBA proxy ovs-albaproxy_data-sv4-02_SV1SRV0001_568a899a-f674-43ce-9649-6c340321a7e6_scrub
Mar 13 10:28:32 SV4SRV0002 celery[32098]: 2018-03-13 10:28:32 01100 -0700 - SV4SRV0002 - 34768/140081113306880 - lib/generic tasks scrub - 5858 - INFO - Scrubber c61e49e1-7708-40f7-829d-80150fd9d811 - vPool data-sv4-02 - StorageRouter SV1SRV0001 - Stack 0 - Creating directory /mnt/hdd1/scrub_work_data-sv4-02_SV1SRV0001_568a899a-f674-43ce-9649-6c340321a7e6
Mar 13 10:28:33 SV4SRV0002 celery[32098]: 2018-03-13 10:28:33 80600 -0700 - SV4SRV0002 - 34768/140081113306880 - extensions/ovs_extensions.generic.volatilemutex - 5859 - WARNING - A lock on ovs_lock_ovs_jobs_scrub_client_cluster_lock was kept for 1.79496097565 sec
Mar 13 10:28:33 SV4SRV0002 celery[32098]: 2018-03-13 10:28:33 80700 -0700 - SV4SRV0002 - 34768/140081113306880 - lib/generic tasks scrub - 5860 - INFO - Scrubber c61e49e1-7708-40f7-829d-80150fd9d811 - vPool data-sv4-02 - StorageRouter SV1SRV0001 - Stack 0 - Deployed ALBA proxy ovs-albaproxy_data-sv4-02_SV1SRV0001_568a899a-f674-43ce-9649-6c340321a7e6_scrub (Config: {u'block_cache': [u'none'], u'log_level': u'info', u'ips': [u'127.0.0.1'], u'manifest_cache_size': 524288000, u'fragment_cache': [u'alba', {u'albamgr_cfg_url': u'arakoon://config/ovs/vpools/08f94559-bc58-454b-9cd7-9ec1ff712e16/proxies/d37f9d7f-2ee9-4d7a-b24d-bfcab2f78d39/config/abm_aa?ini=%2Fopt%2FOpenvStorage%2Fconfig%2Farakoon_cacc.ini', u'cache_on_write': True, u'manifest_cache_size': 524288000, u'cache_on_read': False, u'bucket_strategy': [u'1-to-1', {u'prefix': u'08f94559-bc58-454b-9cd7-9ec1ff712e16', u'preset': u'cache_no_encrypt'}]}], u'read_preference': [], u'albamgr_cfg_url': u'arakoon://config/ovs/vpools/08f94559-bc58-454b-9cd7-9ec1ff712e16/proxies/d37f9d7f-2ee9-4d7a-b24d-bfcab2f78d39/config/abm?ini=%2Fopt%2FOpenvStorage%2Fconfig%2Farakoon_cacc.ini', u'port': 26202, u'transport': 'tcp'})
Mar 13 10:28:33 SV4SRV0002 celery[32098]: 2018-03-13 10:28:33 80700 -0700 - SV4SRV0002 - 34768/140081113306880 - lib/generic tasks scrub - 5861 - INFO - Scrubber c61e49e1-7708-40f7-829d-80150fd9d811 - vPool data-sv4-02 - StorageRouter SV1SRV0001 - Stack 0 - Setting up backend config
Mar 13 10:28:33 SV4SRV0002 celery[32098]: 2018-03-13 10:28:33 81700 -0700 - SV4SRV0002 - 34768/140081113306880 - lib/generic tasks scrub - 5862 - INFO - Scrubber c61e49e1-7708-40f7-829d-80150fd9d811 - vPool data-sv4-02 - StorageRouter SV1SRV0001 - Stack 0 - Backend config was set up
Mar 13 10:28:33 SV4SRV0002 celery[32098]: 2018-03-13 10:28:33 93700 -0700 - SV4SRV0002 - 34768/140081113306880 - lib/generic tasks scrub - 5863 - INFO - Scrubber c61e49e1-7708-40f7-829d-80150fd9d811 - vPool data-sv4-02 - StorageRouter SV1SRV0001 - Stack 0 - Spawning 2 threads for proxy service ovs-albaproxy_data-sv4-02_SV1SRV0001_568a899a-f674-43ce-9649-6c340321a7e6_scrub
Mar 13 10:28:35 SV4SRV0002 celery[32098]: 2018-03-13 10:28:35 77500 -0700 - SV4SRV0002 - 34768/140080613422848 - lib/generic tasks scrub - 5864 - INFO - Scrubber c61e49e1-7708-40f7-829d-80150fd9d811 - vPool data-sv4-02 - StorageRouter SV1SRV0001 - Stack 0 - Scrubbing thread 0 - vDisk pcd_4e4f72f3-bfb7-4b2e-b947-6db828b3e164 - Started scrubbing at location /mnt/hdd1/scrub_work_data-sv4-02_SV1SRV0001_568a899a-f674-43ce-9649-6c340321a7e6
Mar 13 10:28:35 SV4SRV0002 celery[32098]: 2018-03-13 10:28:35 78200 -0700 - SV4SRV0002 - 34768/140080605030144 - lib/generic tasks scrub - 5865 - INFO - Scrubber c61e49e1-7708-40f7-829d-80150fd9d811 - vPool data-sv4-02 - StorageRouter SV1SRV0001 - Stack 0 - Scrubbing thread 1 - vDisk pcd_5144e055-aa0f-4edb-ad54-64761c6d6dec - Started scrubbing at location /mnt/hdd1/scrub_work_data-sv4-02_SV1SRV0001_568a899a-f674-43ce-9649-6c340321a7e6
Mar 13 10:28:35 SV4SRV0002 celery[32098]: 2018-03-13 10:28:35 87000 -0700 - SV4SRV0002 - 34768/140080613422848 - lib/generic tasks scrub - 5866 - INFO - Scrubber c61e49e1-7708-40f7-829d-80150fd9d811 - vPool data-sv4-02 - StorageRouter SV1SRV0001 - Stack 0 - Scrubbing thread 0 - vDisk pcd_4e4f72f3-bfb7-4b2e-b947-6db828b3e164 - Retrieve and apply scrub work
Mar 13 10:28:35 SV4SRV0002 celery[32098]: 2018-03-13 10:28:35 87700 -0700 - SV4SRV0002 - 34768/140080605030144 - lib/generic tasks scrub - 5867 - INFO - Scrubber c61e49e1-7708-40f7-829d-80150fd9d811 - vPool data-sv4-02 - StorageRouter SV1SRV0001 - Stack 0 - Scrubbing thread 1 - vDisk pcd_5144e055-aa0f-4edb-ad54-64761c6d6dec - Retrieve and apply scrub work
....
Mar 13 10:30:40 SV4SRV0002 celery[32098]: 2018-03-13 10:30:40 29600 -0700 - SV4SRV0002 - 34768/140080605030144 - lib/generic tasks scrub - 5883 - ERROR - Scrubber c61e49e1-7708-40f7-829d-80150fd9d811 - vPool data-sv4-02 - StorageRouter SV1SRV0001 - Stack 0 - Scrubbing thread 1 - vDisk pcd_de7052ff-959b-4539-b225-16fba85b3903 - Scrubbing failed
Mar 13 10:30:40 SV4SRV0002 celery[32098]: Traceback (most recent call last):
Mar 13 10:30:40 SV4SRV0002 celery[32098]:   File "/opt/OpenvStorage/ovs/lib/helpers/generic/scrubber.py", line 508, in _execute_scrub
Mar 13 10:30:40 SV4SRV0002 celery[32098]:     backend_config=Configuration.get_configuration_path(self.backend_config_key))
Mar 13 10:30:40 SV4SRV0002 celery[32098]:   File "/usr/lib/python2.7/dist-packages/rpyc/core/netref.py", line 196, in __call__
Mar 13 10:30:40 SV4SRV0002 celery[32098]:     return syncreq(_self, consts.HANDLE_CALL, args, kwargs)
Mar 13 10:30:40 SV4SRV0002 celery[32098]:   File "/usr/lib/python2.7/dist-packages/rpyc/core/netref.py", line 71, in syncreq
Mar 13 10:30:40 SV4SRV0002 celery[32098]:     return conn.sync_request(handler, oid, *args)
Mar 13 10:30:40 SV4SRV0002 celery[32098]:   File "/usr/lib/python2.7/dist-packages/rpyc/core/protocol.py", line 441, in sync_request
Mar 13 10:30:40 SV4SRV0002 celery[32098]:     raise obj
Mar 13 10:30:40 SV4SRV0002 celery[32098]: RuntimeError: scrubber returned with nonzero exit status 256
Mar 13 10:30:40 SV4SRV0002 celery[32098]: ========= Remote Traceback (1) =========
Mar 13 10:30:40 SV4SRV0002 celery[32098]: Traceback (most recent call last):
Mar 13 10:30:40 SV4SRV0002 celery[32098]:   File "/opt/OpenvStorage/tmp.ad53IqJ1U1/rpyc/core/protocol.py", line 305, in _dispatch_request
Mar 13 10:30:40 SV4SRV0002 celery[32098]:     res = self._HANDLERS[handler](self, *args)
Mar 13 10:30:40 SV4SRV0002 celery[32098]:   File "/opt/OpenvStorage/tmp.ad53IqJ1U1/rpyc/core/protocol.py", line 535, in _handle_call
Mar 13 10:30:40 SV4SRV0002 celery[32098]:     return self._local_objects[oid](*args, **dict(kwargs))
Mar 13 10:30:40 SV4SRV0002 celery[32098]: RuntimeError: scrubber returned with nonzero exit status 256
```

scrub.log:
```
Mar 13 10:32:44 SV1SRV0001 alba[10496]: 2018-03-13 10:32:44 663708 -0700 - SV1SRV0001 - 10496/0000 - alba/proxy - 0 - info - Found the followin
g config: {;    "block_cache": [; "none";    ], ;    "log_level": "info", ;    "fragment_cache": [; "none";    ], ;    "manifest_cache_size": 524288000, ;    "ips": [; "127.0.0.1";   ], ;   "read_preference": [], ;    "albamgr_cfg_url": "arakoon://config/ovs/vpools/a99e685c-46f9-4bd4-9017-5355732c2afb/proxies/cab5e9e6-2938-4ec0-8d59-a4e4b0be0cc7/config/abm?ini=%2Fopt%2FOpenvStorage%2Fconfig2Farakoon_cacc.ini", ;    "port": 26202, ;    "transport": "tcp";}
Mar 13 10:32:44 SV1SRV0001 alba[10496]: 2018-03-13 10:32:44 664801 -0700 - SV1SRV0001 - 10496/0000 - alba/proxy - 1 - info - Interpreted the config as: { Cli_proxy.Config.ips = ["127.0.0.1"]; transport = "tcp"; port = 26202;;  log_level = "info"; albamgr_cfg_file = None;;  albamgr_cfg_url =;  (Some "arakoon://config/ovs/vpools/a99e685c-46f9-4bd4-9017-5355732c2afb/proxies/cab5e9e6-2938-4ec0-8d59-a4e4b0be0cc7/config/abm?ini=%2Fopt%2FOpenvStorage%2Fconfig%2Farakoon_cacc.ini");;  fragment_cache = (Some Blob_cache_config.None');;  block_cache = (Some Blob_cache_config.None'); fragment_cache_dir = None;;  fragment_cache_size = None; manifest_cache_size = 524288000;;  albamgr_connection_pool_size = 10; nsm_host_connection_pool_size = 10;;  osd_connection_pool_size = 10; osd_timeout = 10.;;  lwt_preemptive_thread_pool_min_size = 10;;  lwt_preemptive_thread_pool_max_size = 20; chattiness = None;;  max_client_connections = 128; tls_client = None;;  tcp_keepalive =;  { Tcp_keepalive.enable_tcp_keepalive = true; tcp_keepalive_time = 20;;    tcp_keepalive_intvl = 20; tcp_keepalive_probes = 3 };;  use_fadvise = true; upload_slack = 0.2; read_preference = [];;  propagate_osd_info_delay = 60.; fill_bc_from_fc = false; pracc = None;;  partial_read_retry_count = 2; partial_read_retry_interval_seconds = 0.5;;  partial_read_retry_backoff_multiplier = 2.; partial_reconstruct = true }
....
2018-03-13 10:32:44 807836 -0700 - SV1SRV0001 - 9907/0x00007f34a9332900 - scrubber/AlbaConnection - 0000000000000069 - error - convert_exceptions_: read object: caught Alba proxy exception: Proxy_protocol.Protocol.Error.NamespaceDoesNotExist
2018-03-13 10:32:44 807996 -0700 - SV1SRV0001 - 9907/0x00007f34a9332900 - scrubber/BackendConnectionInterfaceLogger - 000000000000006a - error - ~Logger: Exiting read for 124d8f7e-345a-4647-8404-f37dd849ce43 snapshots.xml with exception
2018-03-13 10:32:44 808044 -0700 - SV1SRV0001 - 9907/0x00007f34a9332900 - scrubber/BackendConnectionInterfaceLogger - 000000000000006b - error - ~Logger: Exiting read for 124d8f7e-345a-4647-8404-f37dd849ce43
2018-03-13 10:32:44 808234 -0700 - SV1SRV0001 - 9907/0x00007f34a9332900 - scrubber/FilePool - 000000000000006c - info - ~FilePool: Cleaning up FilePool at /mnt/hdd1/scrub_work_data-sv4-02_SV1SRV0001_568a899a-f674-43ce-9649-6c340321a7e6/f92e3cee-6a60-4551-91e9-858951aeda2a

```

alba.log:
Both configs are for ss-sv4-01. No proxy config found for data-sv4-02.
```
root@SV4SRV0024:/mnt/monitoring/rsyslog/client_logs/SV4SRV0002# zgrep 'Config' ../SV1SRV0001/alba.log.* | grep 26202
../SV1SRV0001/alba.log.1.gz:Mar 13 10:32:44 SV1SRV0001 alba[10496]: 2018-03-13 10:32:44 664801 -0700 - SV1SRV0001 - 10496/0000 - alba/proxy - 1 - info - Interpreted the config as: { Cli_proxy.Config.ips = ["127.0.0.1"]; transport = "tcp"; port = 26202;;  log_level = "info"; albamgr_cfg_file = None;;  albamgr_cfg_url =;  (Some "arakoon://config/ovs/vpools/a99e685c-46f9-4bd4-9017-5355732c2afb/proxies/cab5e9e6-2938-4ec0-8d59-a4e4b0be0cc7/config/abm?ini=%2Fopt%2FOpenvStorage%2Fconfig%2Farakoon_cacc.ini");;  fragment_cache = (Some Blob_cache_config.None');;  block_cache = (Some Blob_cache_config.None'); fragment_cache_dir = None;;  fragment_cache_size = None; manifest_cache_size = 524288000;;  albamgr_connection_pool_size = 10; nsm_host_connection_pool_size = 10;;  osd_connection_pool_size = 10; osd_timeout = 10.;;  lwt_preemptive_thread_pool_min_size = 10;;  lwt_preemptive_thread_pool_max_size = 20; chattiness = None;;  max_client_connections = 128; tls_client = None;;  tcp_keepalive =;  { Tcp_keepalive.enable_tcp_keepalive = true; tcp_keepalive_time = 20;;    tcp_keepalive_intvl = 20; tcp_keepalive_probes = 3 };;  use_fadvise = true; upload_slack = 0.2; read_preference = [];;  propagate_osd_info_delay = 60.; fill_bc_from_fc = false; pracc = None;;  partial_read_retry_count = 2; partial_read_retry_interval_seconds = 0.5;;  partial_read_retry_backoff_multiplier = 2.; partial_reconstruct = true }
../SV1SRV0001/alba.log.2.gz:Mar 12 11:57:27 SV1SRV0001 alba[4865]: 2018-03-12 11:57:27 498758 -0700 - SV1SRV0001 - 4865/0000 - alba/proxy - 1 - info - Interpreted the config as: { Cli_proxy.Config.ips = ["127.0.0.1"]; transport = "tcp"; port = 26202;;  log_level = "info"; albamgr_cfg_file = None;;  albamgr_cfg_url =;  (Some "arakoon://config/ovs/vpools/a99e685c-46f9-4bd4-9017-5355732c2afb/proxies/cab5e9e6-2938-4ec0-8d59-a4e4b0be0cc7/config/abm?ini=%2Fopt%2FOpenvStorage%2Fconfig%2Farakoon_cacc.ini");;  fragment_cache = (Some Blob_cache_config.None');;  block_cache = (Some Blob_cache_config.None'); fragment_cache_dir = None;;  fragment_cache_size = None; manifest_cache_size = 524288000;;  albamgr_connection_pool_size = 10; nsm_host_connection_pool_size = 10;;  osd_connection_pool_size = 10; osd_timeout = 10.;;  lwt_preemptive_thread_pool_min_size = 10;;  lwt_preemptive_thread_pool_max_size = 20; chattiness = None;;  max_client_connections = 128; tls_client = None;;  tcp_keepalive =;  { Tcp_keepalive.enable_tcp_keepalive = true; tcp_keepalive_time = 20;;    tcp_keepalive_intvl = 20; tcp_keepalive_probes = 3 };;  use_fadvise = true; upload_slack = 0.2; read_preference = [];;  propagate_osd_info_delay = 60.; fill_bc_from_fc = false; pracc = None;;  partial_read_retry_count = 2; partial_read_retry_interval_seconds = 0.5;;  partial_read_retry_backoff_multiplier = 2.; partial_reconstruct = true }

```

# Extra Info
Maybe this is not related but i found that the scrub proxy is not registered on the ServiceList.
Is this normal or not?

running proxy:
```
ovs-albaproxy_data-sv4-01_SV1SRV0001_568a899a-f674-43ce-9649-6c340321a7e6_scrub  active  6083
```
list of services:
```
In [20]: for sv in ServiceList.get_services():
    print sv.name
   ....:     
metadataserver_os-sv4-01_0
albaproxy_os-sv4-01_1
arakoon-globalbackend01-nsm_4
metadataserver_data-sv4-01_0
albaproxy_os-sv4-02_3
albaproxy_ss-sv4-01_0
albaproxy_data-sv4-02_5
metadataserver_os-g-01_1
albaproxy_os-sv4-01_2
albaproxy_data-g-02_7
albaproxy_data-sv4-02_7
metadataserver_data-sv4-02_2
albaproxy_data-g-01_0
albaproxy_data-sv4-01_4
albaproxy_data-sv4-01_2
albaproxy_data-g-01_7
albaproxy_data-g-02_2
albaproxy_data-g-02_1
albaproxy_data-g-01_1
albaproxy_ss-sv4-02_2
albaproxy_os-g-01_2
albaproxy_data-g-02_4
metadataserver_data-g-01_0
albaproxy_data-sv4-02_5
metadataserver_data-g-02_0
metadataserver_os-sv4-01_0
albaproxy_data-g-01_6
albaproxy_os-sv4-01_2
albaproxy_data-g-01_2
albaproxy_data-sv4-01_5
albaproxy_data-sv4-01_0
albaproxy_os-sv4-02_2
metadataserver_data-sv4-02_1
albaproxy_data-sv4-01_4
albaproxy_data-g-02_1
albaproxy_data-sv4-01_3
arakoon-globalbackend01-nsm_1
arakoon-globalbackend01-nsm_8
albaproxy_os-g-01_0
albaproxy_data-g-02_5
arakoon-ovsdb
metadataserver_os-sv4-02_0
albaproxy_data-sv4-01_3
albaproxy_data-g-02_7
albaproxy_ss-sv4-02_3
albaproxy_data-g-01_7
metadataserver_os-g-01_0
metadataserver_data-sv4-01_2
albaproxy_data-sv4-01_6
albaproxy_data-g-01_5
albaproxy_data-g-01_1
metadataserver_os-sv4-02_0
metadataserver_data-sv4-02_2
albaproxy_data-g-02_5
albaproxy_ss-sv4-02_1
metadataserver_data-sv4-01_1
arakoon-globalbackend01-nsm_7
albaproxy_data-sv4-02_1
albaproxy_data-g-01_3
albaproxy_os-sv4-01_3
albaproxy_data-g-02_4
albaproxy_os-sv4-01_2
metadataserver_data-g-01_0
albaproxy_ss-sv4-01_1
albaproxy_os-g-01_0
albaproxy_os-sv4-01_0
albaproxy_data-g-02_6
albaproxy_os-g-01_1
metadataserver_data-g-02_0
arakoon-globalbackend01-nsm_2
albaproxy_ss-sv4-01_3
albaproxy_data-g-01_1
albaproxy_ss-sv4-01_1
albaproxy_data-sv4-02_4
albaproxy_os-sv4-01_3
albaproxy_os-sv4-02_0
albaproxy_data-g-01_2
albaproxy_data-sv4-01_1
albaproxy_data-sv4-01_7
albaproxy_data-g-01_3
arakoon-globalbackend01-nsm_9
albaproxy_data-g-02_4
albaproxy_data-g-01_4
albaproxy_data-sv4-01_0
albaproxy_ss-sv4-02_0
albaproxy_data-sv4-02_0
albaproxy_os-sv4-01_3
albaproxy_os-g-01_1
albaproxy_data-g-01_2
metadataserver_data-sv4-01_1
albaproxy_data-g-01_5
albaproxy_ss-sv4-01_2
albaproxy_os-g-01_2
albaproxy_os-sv4-01_1
albaproxy_data-sv4-01_5
metadataserver_data-sv4-02_1
metadataserver_os-sv4-01_1
metadataserver_data-sv4-02_2
albaproxy_data-g-02_3
metadataserver_os-g-01_1
metadataserver_data-sv4-02_0
albaproxy_data-g-01_7
albaproxy_data-g-01_4
albaproxy_os-sv4-02_1
metadataserver_data-g-02_0
metadataserver_ss-sv4-02_0
albaproxy_data-g-01_3
albaproxy_data-sv4-02_6
metadataserver_os-g-01_0
albaproxy_ss-sv4-02_2
metadataserver_ss-sv4-02_0
metadataserver_os-sv4-02_0
albaproxy_data-sv4-02_6
albaproxy_data-sv4-02_3
albaproxy_os-sv4-02_2
albaproxy_data-sv4-02_2
albaproxy_os-g-01_2
metadataserver_data-sv4-02_1
albaproxy_data-g-01_0
metadataserver_os-sv4-01_1
metadataserver_ss-sv4-01_0
albaproxy_data-sv4-01_7
albaproxy_data-g-02_3
albaproxy_data-g-01_0
albaproxy_data-g-02_1
metadataserver_data-g-02_1
albaproxy_os-sv4-02_3
albaproxy_data-g-02_0
albaproxy_data-sv4-02_7
albaproxy_data-g-02_2
albaproxy_data-sv4-02_1
albaproxy_data-sv4-01_0
arakoon-globalbackend01-nsm_5
metadataserver_data-g-02_1
metadataserver_data-sv4-01_2
albaproxy_os-sv4-01_0
albaproxy_data-sv4-01_2
albaproxy_data-g-02_2
albaproxy_os-sv4-02_1
albaproxy_data-sv4-02_5
albaproxy_data-sv4-01_4
albaproxy_os-sv4-01_1
albaproxy_data-sv4-01_1
albaproxy_os-sv4-01_0
metadataserver_data-sv4-01_2
albaproxy_ss-sv4-01_0
metadataserver_data-sv4-02_0
albaproxy_data-sv4-02_0
albaproxy_os-g-01_0
arakoon-globalbackend01-nsm_3
albaproxy_data-sv4-02_2
metadataserver_ss-sv4-01_0
albaproxy_data-g-02_7
albaproxy_data-sv4-01_3
albaproxy_data-sv4-02_3
metadataserver_os-g-01_0
albaproxy_data-sv4-02_4
albaproxy_data-g-02_0
albaproxy_os-g-01_3
albaproxy_ss-sv4-01_2
albaproxy_data-sv4-02_3
metadataserver_os-sv4-01_0
albaproxy_data-g-02_6
albaproxy_os-g-01_3
metadataserver_data-sv4-01_0
albaproxy_data-g-01_6
metadataserver_data-sv4-01_1
albaproxy_os-g-01_1
albaproxy_data-sv4-02_6
albaproxy_os-sv4-02_2
metadataserver_data-sv4-01_0
albaproxy_os-sv4-02_1
albaproxy_data-sv4-01_6
albaproxy_data-sv4-02_2
albaproxy_os-g-01_3
albaproxy_data-sv4-02_1
albaproxy_data-g-01_6
albaproxy_data-g-02_0
metadataserver_data-g-02_1
albaproxy_data-g-01_4
arakoon-globalbackend01-abm
metadataserver_data-g-01_0
albaproxy_os-sv4-02_0
albaproxy_os-sv4-02_0
albaproxy_data-sv4-01_2
arakoon-voldrv
albaproxy_ss-sv4-02_0
albaproxy_ss-sv4-02_3
albaproxy_ss-sv4-01_3
albaproxy_data-g-02_6
albaproxy_data-sv4-02_0
albaproxy_data-sv4-01_7
albaproxy_data-sv4-01_1
metadataserver_data-sv4-02_0
albaproxy_data-sv4-01_6
metadataserver_os-sv4-01_1
albaproxy_ss-sv4-02_1
albaproxy_data-g-02_5
albaproxy_data-sv4-02_7
albaproxy_os-sv4-02_3
albaproxy_data-g-02_3
albaproxy_data-sv4-01_5
albaproxy_data-sv4-02_4
metadataserver_os-g-01_1
albaproxy_data-g-01_5
arakoon-globalbackend01-nsm_6
arakoon-globalbackend01-nsm_0
```

**Comment by JeffreyDevloo** [comment]
**Answers to your questions:**
`So how do you guys check if a scrub proxy has been deployed and is running?
Do you only check if the port is online or do you check via the service manager if the process is running?`
We check the status in Systemd

These logs indicate that the scrubbing proxy was created by this job.
```
Mar 13 10:28:32 SV4SRV0002 celery[32098]: 2018-03-13 10:28:31 100000 -0700 - SV4SRV0002 - 34768/140081803060992 - lib/generic tasks scrub - 5853 - INFO - Scrubber c61e49e1-7708-40f7-829d-80150fd9d811 - vPool data-sv4-02 - Spawning 1 stack
Mar 13 10:28:32 SV4SRV0002 celery[32098]: 2018-03-13 10:28:32 00600 -0700 - SV4SRV0002 - 34768/140081113306880 - lib/generic tasks scrub - 5854 - INFO - Scrubber c61e49e1-7708-40f7-829d-80150fd9d811 - vPool data-sv4-02 - StorageRouter SV1SRV0001 - Stack 0 - Checking for ALBA proxy ovs-albaproxy_data-sv4-02_SV1SRV0001_568a899a-f674-43ce-9649-6c340321a7e6_scrub deployment
Mar 13 10:28:32 SV4SRV0002 celery[32098]: 2018-03-13 10:28:32 00600 -0700 - SV4SRV0002 - 34768/140081113306880 - lib/generic tasks scrub - 5855 - INFO - Scrubber c61e49e1-7708-40f7-829d-80150fd9d811 - vPool data-sv4-02 - StorageRouter SV1SRV0001 - Stack 0 - Registering usage of ovs-albaproxy_data-sv4-02_SV1SRV0001_568a899a-f674-43ce-9649-6c340321a7e6_scrub
Mar 13 10:28:32 SV4SRV0002 celery[32098]: 2018-03-13 10:28:32 00900 -0700 - SV4SRV0002 - 34768/140081113306880 - lib/generic tasks scrub - 5856 - INFO - Scrubber c61e49e1-7708-40f7-829d-80150fd9d811 - vPool data-sv4-02 - StorageRouter SV1SRV0001 - Stack 0 - Current registered users: [{'stack_id': '514d1ec9-9ff5-447c-991c-9a7a6caa2467', 'storagerouter_guid': '3579dde2-a7c2-4e4f-a24e-295dadd3a0f6', 'job_id': 'c61e49e1-7708-40f7-829d-80150fd9d811', 'worker_pid': 32098, 'worker_start': 'Mon Mar 12 14:19:18 2018'}]
Mar 13 10:28:32 SV4SRV0002 celery[32098]: 2018-03-13 10:28:32 01000 -0700 - SV4SRV0002 - 34768/140081113306880 - lib/generic tasks scrub - 5857 - INFO - Scrubber c61e49e1-7708-40f7-829d-80150fd9d811 - vPool data-sv4-02 - StorageRouter SV1SRV0001 - Stack 0 - Deploying ALBA proxy ovs-albaproxy_data-sv4-02_SV1SRV0001_568a899a-f674-43ce-9649-6c340321a7e6_scrub
Mar 13 10:28:32 SV4SRV0002 celery[32098]: 2018-03-13 10:28:32 01100 -0700 - SV4SRV0002 - 34768/140081113306880 - lib/generic tasks scrub - 5858 - INFO - Scrubber c61e49e1-7708-40f7-829d-80150fd9d811 - vPool data-sv4-02 - StorageRouter SV1SRV0001 - Stack 0 - Creating directory /mnt/hdd1/scrub_work_data-sv4-02_SV1SRV0001_568a899a-f674-43ce-9649-6c340321a7e6
Mar 13 10:28:33 SV4SRV0002 celery[32098]: 2018-03-13 10:28:33 80600 -0700 - SV4SRV0002 - 34768/140081113306880 - extensions/ovs_extensions.generic.volatilemutex - 5859 - WARNING - A lock on ovs_lock_ovs_jobs_scrub_client_cluster_lock was kept for 1.79496097565 sec
Mar 13 10:28:33 SV4SRV0002 celery[32098]: 2018-03-13 10:28:33 80700 -0700 - SV4SRV0002 - 34768/140081113306880 - lib/generic tasks scrub - 5860 - INFO - Scrubber c61e49e1-7708-40f7-829d-80150fd9d811 - vPool data-sv4-02 - StorageRouter SV1SRV0001 - Stack 0 - Deployed ALBA proxy ovs-albaproxy_data-sv4-02_SV1SRV0001_568a899a-f674-43ce-9649-6c340321a7e6_scrub (Config: {u'block_cache': [u'none'], u'log_level': u'info', u'ips': [u'127.0.0.1'], u'manifest_cache_size': 524288000, u'fragment_cache': [u'alba', {u'albamgr_cfg_url': u'arakoon://config/ovs/vpools/08f94559-bc58-454b-9cd7-9ec1ff712e16/proxies/d37f9d7f-2ee9-4d7a-b24d-bfcab2f78d39/config/abm_aa?ini=%2Fopt%2FOpenvStorage%2Fconfig%2Farakoon_cacc.ini', u'cache_on_write': True, u'manifest_cache_size': 524288000, u'cache_on_read': False, u'bucket_strategy': [u'1-to-1', {u'prefix': u'08f94559-bc58-454b-9cd7-9ec1ff712e16', u'preset': u'cache_no_encrypt'}]}], u'read_preference': [], u'albamgr_cfg_url': u'arakoon://config/ovs/vpools/08f94559-bc58-454b-9cd7-9ec1ff712e16/proxies/d37f9d7f-2ee9-4d7a-b24d-bfcab2f78d39/config/abm?ini=%2Fopt%2FOpenvStorage%2Fconfig%2Farakoon_cacc.ini', u'port': 26202, u'transport': 'tcp'})
Mar 13 10:28:33 SV4SRV0002 celery[32098]: 2018-03-13 10:28:33 80700 -0700 - SV4SRV0002 - 34768/140081113306880 - lib/generic tasks scrub - 5861 - INFO - Scrubber c61e49e1-7708-40f7-829d-80150fd9d811 - vPool data-sv4-02 - StorageRouter SV1SRV0001 - Stack 0 - Setting up backend config
Mar 13 10:28:33 SV4SRV0002 celery[32098]: 2018-03-13 10:28:33 81700 -0700 - SV4SRV0002 - 34768/140081113306880 - lib/generic tasks scrub - 5862 - INFO - Scrubber c61e49e1-7708-40f7-829d-80150fd9d811 - vPool data-sv4-02 - StorageRouter SV1SRV0001 - Stack 0 - Backend config was set up
```

Since it could start, that means that the port was not in use. So either the logging is irrelevant or something else is going on.

~The reason as to why you might not have found the service or the config anymore is because it might have been deleted by the scrubber.~ The scrubbing services are not saved in the DAL

Another thing is: `Mar 13 10:28:31 SV4SRV0002 celery[32098]: 2018-03-13 10:28:31 47200 -0700 - SV4SRV0002 - 34768/140081803060992 - lib/generic tasks scrub - 5705 - INFO - Scrubber c61e49e1-7708-40f7-829d-80150fd9d811 - vPool data-sv4-02 - Will be removing {u'worker_pid': 26225, u'vdisk_guid': u'52457ddd-439e-4254-aa4d-e4b86e16fad1', u'storagerouter_guid': u'26955ac4-d6cb-4abc-9536-bf1b74527b32', u'worker_start': u'Mon Mar  5 13:26:46 2018'} on the next save as it is no longer relevant` indicates that a previous scrubjob was killed so it might have something to do with this

**Comment by jeroenmaelbrancke** [comment]
Maybe because the workers are getting killed with a signal 15 (#1992)?

**Comment by JeffreyDevloo** [comment]
Looks like the issue could be that a second scrub job would have deployed a new proxy and the older proxy (of the job for which you provided logging) might have died, causing the port to free up. This should all have happened in the 2 minute timewindow before the volumedriver was starting to scrub.

We should protect ourselves against this though.
The real question is why the proxy did not start/use the port that we specified...

**Comment by wimpers** [comment]
Since this is a corner case, set to roadmap. Also, seen only once.

**Comment by JeffreyDevloo** [comment]
It was seen again

**Comment by wimpers** [comment]
Any idea what was going on earlier or do we have no clue why?

**Comment by JeffreyDevloo** [comment]
The configuration passed is based on port. There might be a particular scenario where the port points towards a proxy that got removed in meantime but the client was still able to connect.

The details are not clear yet but writing some extra tests around port allocation might expose the issue.

**Comment by jeroenmaelbrancke** [comment]
It was seen again on Nuvolat:

proxy config:
```
{;
"block_cache": [;
    "none";
], ;
"log_level": "info", ;
"osd_connection_pool_size": 3, ;
"ips": [;
    "127.0.0.1";
], ;
"manifest_cache_size": 524288000, ;
"fragment_cache": [;
    "alba", ;
    {;
        "albamgr_cfg_url": "arakoon://config/ovs/vpools/fce6a466-cf79-4d32-bd3a-b435f9e70440/proxies/b49379ac-d1bc-498b-9a73-398939dbf85d/config/abm_aa?ini=%2Fopt%2FOpenvStorage%2Fconfig%2Farakoon_cacc.ini", ;
        "cache_on_write": true, ;
        "manifest_cache_size": 524288000, ;
        "cache_on_read": false, ;
        "bucket_strategy": [;
            "1-to-1", ;
            {;
                "prefix": "fce6a466-cf79-4d32-bd3a-b435f9e70440", ;
                "preset": "cache_no_encrypt";
            };
        ];
    };
], ;
"read_preference": [;
    "e4a9beee-2eff-466a-951d-257bf8395a0a";
], ;
"albamgr_cfg_url": "arakoon://config/ovs/vpools/fce6a466-cf79-4d32-bd3a-b435f9e70440/proxies/b49379ac-d1bc-498b-9a73-398939dbf85d/config/abm?ini=%2Fopt%2FOpenvStorage%2Fconfig%2Farakoon_cacc.ini", ;
"port": 26201, ;
"transport": "tcp";}
Aug 28 05:35:53 NY1SRV1001 alba[1411]: 2018-08-28 05:35:53 937176 -0400 - NY1SRV1001 - 1411/0000 - alba/proxy - 1 - info - Interpreted the config as: { ips = ["127.0.0.1"]; transport = "tcp"; port = 26201; log_level = "info";;  albamgr_cfg_file = None;;  albamgr_cfg_url =;  (Some "arakoon://config/ovs/vpools/fce6a466-cf79-4d32-bd3a-b435f9e70440/proxies/b49379ac-d1bc-498b-9a73-398939dbf85d/config/abm?ini=%2Fopt%2FOpenvStorage%2Fconfig%2Farakoon_cacc.ini");;  fragment_cache =;  (Some (Alba;
       { albamgr_cfg_url =;
         "arakoon://config/ovs/vpools/fce6a466-cf79-4d32-bd3a-b435f9e70440/proxies/b49379ac-d1bc-498b-9a73-398939dbf85d/config/abm_aa?ini=%2Fopt%2FOpenvStorage%2Fconfig%2Farakoon_cacc.ini";;
         bucket_strategy =;
         (Blob_cache_alba.OneOnOne;
            { Blob_cache_alba.preset = "cache_no_encrypt";;
              prefix = "fce6a466-cf79-4d32-bd3a-b435f9e70440" });;
         fragment_cache = None'; manifest_cache_size = 524288000;;
         albamgr_connection_pool_size = 10;;
         nsm_host_connection_pool_size = 10;;
         osd_connection_pool_size = 10; osd_timeout = 2.;;
         tls_client = None; cache_on_read_ = false;;
         cache_on_write_ = true; block_size_ = None }));;
```

Proxy error:
```
Aug 28 05:37:13 NY1SRV1001 alba[1316]: 2018-08-28 05:37:13 592002 -0400 - NY1SRV1001 - 1316/0000 - alba/proxy - 479 - info - Nsm_host_access.with_namespace_id: namespace "1971bbc7-c646-4813-9599-e052d546a60c" was not found
```

Namespace details:
```
In [11]: vd = VDiskList.get_vdisk_by_volume_id('1971bbc7-c646-4813-9599-e052d546a60c')
In [12]: vd.name
Out[12]: u'pcd_350792ea-4355-4d90-9b66-a9b16c996298'
In [14]: vd.vpool.name
Out[14]: u'data-ny1-06'
In [15]: vd.guid
Out[15]: '04c9101e-96cf-480a-8463-ab457b546d41'
In [16]: vd.vpool.guid
Out[16]: 'd6e12a8f-79ab-45c5-a4b3-4922ce841a3a'
```

The albamgr_cfg_url in the proxy config is not the one of vpool data-ny1-06 but from data-g-06.

**Comment by JeffreyDevloo** [comment]
Unable to determine which scrub job this was from and what config it got.
I'd need the proxy deployment logs:
```
/mnt/monitoring/rsyslog/client_logs/NY2SRV0004/ovs-workers.log:Aug 28 07:50:43 NY2SRV0004 ovs-workers[39802]: 2018-08-28 07:50:43 46000 -0400 - NY2SRV0004 - 39817/140077140731648 - lib/generic tasks scrub - 6095 - INFO - Scrubber 9cdd50b5-0339-4066-b685-b8f0e32cd2f5 - vPool ss-ny2-01 - StorageRouter NY1SRV1000 - Stack 0 - Deployed ALBA proxy ovs-albaproxy_ss-ny2-01_NY1SRV1000_f66d3c02-3748-4d3e-a989-6b5e0db1b235_scrub_0 (Config: {u'block_cache': [u'none'], u'log_level': u'info', u'osd_connection_pool_size': 3, u'fragment_cache': [u'none'], u'albamgr_connection_pool_size': 5, u'manifest_cache_size': 524288000, u'nsm_host_connection_pool_size': 5, u'ips': [u'127.0.0.1'], u'read_preference': [], u'albamgr_cfg_url': u'arakoon://config/ovs/vpools/4fca8034-83a1-46fb-b573-4200886b95e6/proxies/fb5ff0d7-d2b2-40b8-b077-a059fe7bbdda/config/abm?ini=%2Fopt%2FOpenvStorage%2Fconfig%2Farakoon_cacc.ini', u'port': 26216, u'transport': 'tcp'})
```
And the backend config passed. The key for this backend config can be computed using the proxy name.
Example: ovs-albaproxy_ss-ny2-01_NY1SRV1000_f66d3c02-3748-4d3e-a989-6b5e0db1b235_scrub_0. This one shows the vpool name (ss-ny2-01), storagerouter name (NY1SRV1000) and the guid of the scrub partition (f66d3c02-3748-4d3e-a989-6b5e0db1b235)
Using the guid of the vpool and the guid of the scrub partition: the backend can be retrieved
```
In [10]: !ovs config get ovs/vpools/4fca8034-83a1-46fb-b573-4200886b95e6/proxies/scrub/backend_config_f66d3c02-3748-4d3e-a989-6b5e0db1b235
{
    "backend_connection_manager": {
        "backend_interface_retries_on_error": 120, 
        "backend_interface_partial_read_retry_backoff_multiplier": 2.0, 
        "backend_type": "MULTI", 
        "backend_interface_partial_read_timeout_max_msecs": 30000, 
        "backend_connection_pool_capacity": 32, 
        "backend_interface_retry_backoff_multiplier": 1.7, 
        "backend_interface_partial_read_retry_interval_msecs": 100, 
        "1": {
            "alba_connection_preset": "local_ss_encrypt", 
            "backend_type": "ALBA", 
            "alba_connection_asd_connection_pool_capacity": 10, 
            "alba_connection_transport": "TCP", 
            "alba_connection_use_rora": true, 
            "alba_connection_timeout": 30, 
            "alba_connection_host": "127.0.0.1", 
            "alba_connection_rora_manifest_cache_capacity": 25000, 
            "alba_connection_rora_timeout_msecs": 2000, 
            "alba_connection_port": 26217
        }, 
        "0": {
            "alba_connection_preset": "local_ss_encrypt", 
            "backend_type": "ALBA", 
            "alba_connection_asd_connection_pool_capacity": 10, 
            "alba_connection_transport": "TCP", 
            "alba_connection_use_rora": true, 
            "alba_connection_timeout": 30, 
            "alba_connection_host": "127.0.0.1", 
            "alba_connection_rora_manifest_cache_capacity": 25000, 
            "alba_connection_rora_timeout_msecs": 2000, 
            "alba_connection_port": 26216
        }, 
        "3": {
            "alba_connection_preset": "local_ss_encrypt", 
            "backend_type": "ALBA", 
            "alba_connection_asd_connection_pool_capacity": 10, 
            "alba_connection_transport": "TCP", 
            "alba_connection_use_rora": true, 
            "alba_connection_timeout": 30, 
            "alba_connection_host": "127.0.0.1", 
            "alba_connection_rora_manifest_cache_capacity": 25000, 
            "alba_connection_rora_timeout_msecs": 2000, 
            "alba_connection_port": 26219
        }, 
        "backend_interface_retry_interval_secs": 5, 
        "backend_interface_partial_read_timeout_multiplier": 8, 
        "backend_interface_partial_read_retries_on_error": 10, 
        "2": {
            "alba_connection_preset": "local_ss_encrypt", 
            "backend_type": "ALBA", 
            "alba_connection_asd_connection_pool_capacity": 10, 
            "alba_connection_transport": "TCP", 
            "alba_connection_use_rora": true, 
            "alba_connection_timeout": 30, 
            "alba_connection_host": "127.0.0.1", 
            "alba_connection_rora_manifest_cache_capacity": 25000, 
            "alba_connection_rora_timeout_msecs": 2000, 
            "alba_connection_port": 26218
        }, 
        "backend_interface_timeout_max_msecs": 3600000, 
        "backend_interface_partial_read_timeout_msecs": 5000
    }
}
```
If these ports mismatch, the scrubber will connect to the wrong proxy.
If so, I'd like to see the logs of the scrubber that deployed those proxies and their configs.

**Comment by jeroenmaelbrancke** [comment]
Found another problem with the scrub proxies.
I started a scrub process for vpool data-g-03 and saw following error messages in the scrubber.log:
```
2018-09-26 03:34:56 728755 -0400 - NY1SRV1001 - 17641/0x00007fe43b232900 - scrubber/BackendConnectionPool - 000000000000066d - warning - error: AlbaConfig{host=127.0.0.1,port=26203,timeout=30,transport=TCP,use_rora=true,rora_manifest_cache_capacity=25000,rora_use_nullio=false,asd_connection_pool_capacity=10,rora_timeout_msecs=50,preset=global-encrypt,is_local=true}: blacklisted for 15000000000 nanoseconds
```

So i had a look if each proxy was running properly:
```
ovs-albaproxy_data-g-03_NY1SRV1001_aa58fc56-07d9-494e-89b0-c0f894749566_scrub_0          active  3355
ovs-albaproxy_data-g-03_NY1SRV1001_aa58fc56-07d9-494e-89b0-c0f894749566_scrub_1          active  3890
ovs-albaproxy_data-g-03_NY1SRV1001_aa58fc56-07d9-494e-89b0-c0f894749566_scrub_2          active  3457
ovs-albaproxy_data-g-03_NY1SRV1001_aa58fc56-07d9-494e-89b0-c0f894749566_scrub_3          active  3990
```

But when i had a look on which ports these proxies were running:
```
root@NY1SRV1001:~# netstat -tulpn | grep alba
tcp        0      0 127.0.0.1:26232         0.0.0.0:*               LISTEN      3355/alba.native
tcp        0      0 127.0.0.1:26233         0.0.0.0:*               LISTEN      3890/alba.native
tcp        0      0 127.0.0.1:26234         0.0.0.0:*               LISTEN      3457/alba.native
tcp        0      0 127.0.0.1:26235         0.0.0.0:*               LISTEN      3990/alba.native
```

None of them use port 26203... which is strange...

All the process were running since September 25 with another config.
```
root@NY1SRV1001:~# ps aux | grep 3355
root      3355  1.3  1.2 2973092 2541280 ?     Ssl  Sep25   9:52 /opt/alba-1.5.30/bin/alba.native proxy-start --config arakoon://config/ovs/vpools/7b08fcbf-91f6-4a29-9d11-19536721192a/proxies/scrub/scrub_config_aa58fc56-07d9-494e-89b0-c0f894749566_0?ini=%2Fopt%2FOpenvStorage%2Fconfig%2Farakoon_cacc.ini --log-sink console:
root     20389  0.0  0.0  14224  1096 pts/0    S+   03:42   0:00 grep --color=auto 3355
root@NY1SRV1001:~# ps aux | grep 3890
root      3890  1.4  2.3 5151580 4667480 ?     Ssl  Sep25  10:37 /opt/alba-1.5.30/bin/alba.native proxy-start --config arakoon://config/ovs/vpools/7b08fcbf-91f6-4a29-9d11-19536721192a/proxies/scrub/scrub_config_aa58fc56-07d9-494e-89b0-c0f894749566_1?ini=%2Fopt%2FOpenvStorage%2Fconfig%2Farakoon_cacc.ini --log-sink console:
root     20392  0.0  0.0  14224  1016 pts/0    S+   03:42   0:00 grep --color=auto 3890
root@NY1SRV1001:~# ps aux | grep 3457
root      3457  1.3  0.8 2149584 1717588 ?     Ssl  Sep25   9:55 /opt/alba-1.5.30/bin/alba.native proxy-start --config arakoon://config/ovs/vpools/7b08fcbf-91f6-4a29-9d11-19536721192a/proxies/scrub/scrub_config_aa58fc56-07d9-494e-89b0-c0f894749566_2?ini=%2Fopt%2FOpenvStorage%2Fconfig%2Farakoon_cacc.ini --log-sink console:
root     20398  0.0  0.0  14224   948 pts/0    S+   03:42   0:00 grep --color=auto 3457
```

After restarting the proxies the correct config was used and the scrubber was able to write/read via the proxy.

So maybe it's a good idea when you deploy a new proxy config to restart the service a second time. If the service was not properly cleaned by the previous run, the current run will always have issues until OPS restart the services.
